client/src/modules/list/components/Items/model/actions.js

```diff
@@ -0,0 +1,188 @@
+  dispatch(addItemRequest());
+  return postData(`${ENDPOINT_URL}/lists/add-item`, { item, listId })
+    .then(resp => resp.json())
+    .then(json => dispatch(addItemSuccess(json, listId)))
+    .catch(err => {
+      dispatch(addItemFailure());
+      createNotificationWithTimeout(
+        dispatch,
+        NotificationType.ERROR,
+        err.message || "Oops, we're sorry, adding item failed..."
+      );
+    });
+};
+
+export const toggle = (
+  isOrdered,
+  itemId,
+  listId,
+  updatedAuthor
+) => dispatch => {
+  dispatch(toggleItemRequest());
+  return patchData(`${ENDPOINT_URL}/lists/${listId}/update-item`, {
+    itemId,
+    isOrdered: !isOrdered
+  })
+    .then(resp => resp.json())
+    .then(item =>
+      setTimeout(() => dispatch(toggleItemSuccess(item, listId)), 600)
+    )
+    .catch(err => {
+      dispatch(toggleItemFailure());
+      createNotificationWithTimeout(
+        dispatch,
+        NotificationType.ERROR,
+        err.message || "Oops, we're sorry, changing item's status failed..."
+      );
+    });
+};
+
+export const setVote = (itemId, listId) => dispatch => {
+  dispatch(voteForItemRequest());
+
+  return patchData(`${ENDPOINT_URL}/lists/${listId}/set-vote`, { itemId })
+    .then(resp => resp.json())
+    .then(item => dispatch(voteForItemSuccess(item, listId)))
+    .catch(err => {
+      dispatch(voteForItemFailure());
+      createNotificationWithTimeout(
+        dispatch,
+        NotificationType.ERROR,
+        err.message || "Oops, we're sorry, voting failed...."
+      );
+    });
+};
+
+export const clearVote = (itemId, listId) => dispatch => {
+  dispatch(voteForItemRequest());
+
+  return patchData(`${ENDPOINT_URL}/lists/${listId}/clear-vote`, { itemId })
+    .then(resp => resp.json())
+    .then(item => dispatch(voteForItemSuccess(item, listId)))
+    .catch(err => {
+      dispatch(voteForItemFailure());
+      createNotificationWithTimeout(
+        dispatch,
+        NotificationType.ERROR,
+        err.message || "Oops, we're sorry, voting failed...."
+      );
+    });
+};
+
+export const updateItemDetails = (listId, itemId, data) => dispatch => {
+  dispatch(updateItemDetailsRequest());
+  return patchData(`${ENDPOINT_URL}/lists/${listId}/update-item-details`, {
+    ...data,
+    itemId
+  })
+    .then(resp => resp.json())
+    .then(json => {
+      dispatch(updateItemDetailsSuccess(listId, itemId, data));
+      createNotificationWithTimeout(
+        dispatch,
+        NotificationType.SUCCESS,
+        json.message
+      );
+    })
+    .catch(err => {
+      dispatch(updateItemDetailsFailure());
+      createNotificationWithTimeout(
+        dispatch,
+        NotificationType.ERROR,
+        err.message
+      );
+    });
+};
+
+export const cloneItem = (listId, itemId) => dispatch => {
+  dispatch(cloneItemRequest());
+  return patchData(`${ENDPOINT_URL}/lists/${listId}/clone-item`, {
+    itemId
+  })
+    .then(resp => resp.json())
+    .then(data => {
```

Comment:
follow the same pattern as we did in previous functions 
so `json`   -->
`dispatch(cloneItemSuccess(listId, data.item));`